### PR TITLE
feat(spells): do json manipulations internally

### PR DESCRIPTION
**Pull request by justprosh**
- spell's `getDataSrv` returns parsed json value, so there is no need to parse it manually
- `spell.install` now accepts any object as `init_data` (previously, it was accepting only stringified ones)
- spell's `response` accepts objects (previously, it was accepting only stringified ones)

### CODE REVIEW

File: sorcerer/src/spells.rs [hunk 1]
```diff
@@ -66,7 +66,7 @@ pub(crate) async fn spell_install(
         services.call_function(
             &spell_id,
             "set_json_fields",
-            vec![json!(init_data)],
+            vec![json!(init_data.to_string())],
```
Comment (folex) [inline sorcerer/src/spells.rs:69]: shouldn't it be enough to pass `init_data.to_string()`? If not, why not?

File: sorcerer/src/spells.rs [hunk 2]
```diff
@@ -262,12 +263,12 @@ pub(crate) fn store_response(
...
 ) -> Result<(), JError> {
     let spell_id = parse_spell_id_from(&params.id)?;
-
+    let response: JValue = Args::next("spell_id", &mut args.function_args.into_iter())?;
     process_func_outcome::<UnitValue>(
         services.call_function(
             spell_id,
             "set_json_fields",
-            args.function_args.clone(),
+            vec![json!(response.to_string())],
```
Comment (folex) [inline sorcerer/src/spells.rs:271]: same thing, why do you serialize it to json twice?

File: sorcerer/src/spells.rs (hunk 1 above)
Comment (justprosh) [reply sorcerer/src/spells.rs:69]: call_function accepts `function_args: Vec<JValue>`

File: sorcerer/src/spells.rs (hunk 2 above)
Comment (folex) [reply sorcerer/src/spells.rs:271]: `response.to_string()` seems to satisfy that description

Comment (justprosh) [reply sorcerer/src/spells.rs:271]: yeah, but `call_function` accepts `function_args: Vec<JValue>`

**Review by folex — APPROVED** [review]

**Pull request merged**